eclair-core/src/test/scala/fr/acinq/eclair/transactions/TransactionsSpec.scala

```diff
@@ -342,41 +339,51 @@ class TransactionsSpec extends FunSuite with Logging {
-    val htlc3 = UpdateAddHtlc(ByteVector32.Zeroes, 1, millibtc2satoshi(MilliBtc(200)).toMilliSatoshi, sha256(paymentPreimage3), CltvExpiry(300), TestConstants.emptyOnionPacket)
+    val htlc1 = UpdateAddHtlc(ByteVector32.Zeroes, 1, millibtc2satoshi(MilliBtc(100)).toMilliSatoshi, sha256(paymentPreimage1), CltvExpiry(300), TestConstants.emptyOnionPacket)
+    val htlc2 = UpdateAddHtlc(ByteVector32.Zeroes, 2, millibtc2satoshi(MilliBtc(200)).toMilliSatoshi, sha256(paymentPreimage2), CltvExpiry(300), TestConstants.emptyOnionPacket)
+    val htlc3 = UpdateAddHtlc(ByteVector32.Zeroes, 3, millibtc2satoshi(MilliBtc(200)).toMilliSatoshi, sha256(paymentPreimage3), CltvExpiry(300), TestConstants.emptyOnionPacket)
+    val htlc4 = UpdateAddHtlc(ByteVector32.Zeroes, 4, millibtc2satoshi(MilliBtc(200)).toMilliSatoshi, sha256(paymentPreimage3), CltvExpiry(300), TestConstants.emptyOnionPacket)
+    val htlc5 = UpdateAddHtlc(ByteVector32.Zeroes, 5, millibtc2satoshi(MilliBtc(200)).toMilliSatoshi, sha256(paymentPreimage3), CltvExpiry(301), TestConstants.emptyOnionPacket)
+
     val spec = CommitmentSpec(
       htlcs = Set(
         DirectedHtlc(IN, htlc1),
         DirectedHtlc(IN, htlc2),
-        DirectedHtlc(IN, htlc3)
+        DirectedHtlc(IN, htlc3),
+        DirectedHtlc(IN, htlc4),
+        DirectedHtlc(IN, htlc5)
       ),
       feeratePerKw = feeratePerKw,
       toLocal = millibtc2satoshi(MilliBtc(400)).toMilliSatoshi,
       toRemote = millibtc2satoshi(MilliBtc(300)).toMilliSatoshi)
 
     val commitTxNumber = 0x404142434446L
-    val commitTx = {
-      val outputs = makeCommitTxOutputs(true, localDustLimit, localRevocationPriv.publicKey, CltvExpiryDelta(toLocalDelay), localDelayedPaymentPriv.publicKey, remotePaymentPriv.publicKey, localHtlcPriv.publicKey, remoteHtlcPriv.publicKey, spec)
+    val (commitTx, outputs) = {
+      val outputs = makeCommitTxOutputs(true, localDustLimit, localRevocationPriv.publicKey, toLocalDelay, localDelayedPaymentPriv.publicKey, remotePaymentPriv.publicKey, localHtlcPriv.publicKey, remoteHtlcPriv.publicKey, spec)
       val txinfo = makeCommitTx(commitInput, commitTxNumber, localPaymentPriv.publicKey, remotePaymentPriv.publicKey, true, outputs)
       val localSig = Transactions.sign(txinfo, localPaymentPriv)
       val remoteSig = Transactions.sign(txinfo, remotePaymentPriv)
-      Transactions.addSigs(txinfo, localFundingPriv.publicKey, remoteFundingPriv.publicKey, localSig, remoteSig)
+      (Transactions.addSigs(txinfo, localFundingPriv.publicKey, remoteFundingPriv.publicKey, localSig, remoteSig), outputs)
     }
 
-    val htlcOut1 :: htlcOut2 :: htlcOut3 :: _ = commitTx.tx.txOut.toList
+    // htlc1 comes before htlc2 because of the smaller amount (BIP69)
+    // htlc2 and htlc3 have the same amount but htlc2 comes first because its pubKeyScript is lexicographically smaller than htlc3's
+    // htlc5 comes before htlc3 and htlc4 because of the smaller CLTV
```

Comment:
That looks surprising...htlc5 has a bigger cltv expiry than htlc3 and htlc4, why does it come first?

Reply:
Got it, there were two issues: we were using incoming HTLCs instead of outgoing.
I have a small fix in https://github.com/ACINQ/eclair/commit/1341652cc7745046f7e5482865dc32719f822c7f if you want to take a look!

Reply:
Oh indeed! I applied both your suggested changes in the latest commit 517ed54